business/src/main/java/org/mskcc/cbio/portal/persistence/ClinicalFieldMapper.java

```diff
@@ -30,4 +31,8 @@
 	List<DBClinicalField> getAllClinicalFields();
    List<DBClinicalField> getAllClinicalFieldsByStudy(@Param("study_id") Integer study_id);
 	List<DBClinicalField> getClinicalFieldsById(@Param("attr_ids") List<String> attr_ids);
+    
+    List<DBClinicalFieldWithStats> getSampleClinicalFieldsByStudyWithStats(@Param("study_id") Integer study_id);
+    List<DBClinicalFieldWithStats> getPatientClinicalFieldsByStudyWithStats(@Param("study_id") Integer study_id);
```

Comment:
check with current oncoprint frontend code whether it expects 2 separate calls or if you should provide 1 integrated call that returns patient + sample attributes.